bin/src/hickory-dns.rs

```diff
@@ -333,147 +366,182 @@ fn main() {
         default();
     }
 
-    info!("Hickory DNS {} starting", hickory_client::version());
-    // start up the server for listening
+    info!("Hickory DNS {} starting...", hickory_client::version());
+
+    // Load configuration files
 
     let config = args.config.clone();
     let config_path = Path::new(&config);
-    info!("loading configuration from: {:?}", config_path);
-    let config = Config::read_config(config_path)
-        .unwrap_or_else(|e| panic!("could not read config {}: {:?}", config_path.display(), e));
+
+    info!("Loading configuration from: {config_path:?}");
+
+    let config = Config::read_config(config_path).expect("Valid configuration files");
     let directory_config = config.get_directory().to_path_buf();
     let zonedir = args.zonedir.clone();
     let zone_dir: PathBuf = zonedir
         .as_ref()
         .map(PathBuf::from)
         .unwrap_or_else(|| directory_config.clone());
 
-    // TODO: allow for num threads configured...
-    let mut runtime = runtime::Builder::new_multi_thread()
-        .enable_all()
-        .worker_threads(4)
-        .thread_name("hickory-server-runtime")
-        .build()
-        .expect("failed to initialize Tokio Runtime");
+    let mut runtime = runtime::Builder::new_multi_thread();
+    runtime.enable_all().thread_name("hickory-server-runtime");
+    if let Some(workers) = args.workers {
+        runtime.worker_threads(workers);
+    }
+    let mut runtime = runtime.build().expect("Running Tokio runtime");
```

Comment:
Yes, it is. The expect messages are encouraged to be positive as the result will return `Ok`. The error message returned by the result indicates the actual trouble happened. If we need to be more specific about the error message, we can use unwrap or else that it may not be the case here.

Documentation:

https://doc.rust-lang.org/std/result/enum.Result.html#method.expect

Reply:
Assume that the tokio runtime panics at `.build()` method, here is the message we expect:

```txt
thread 'main' panicked at bin/src/hickory-dns.rs:395:17:
Running Tokio runtime: ooops, something went wrong here. (the error message returned by tokio).
```

If the error print style is still not clear here,  please let me know to change it  with more verbose `unwrap_or` and `panic!` style.

Reply:
Yes, I still think "failed to initialize Tokio Runtime: oops, something went wrong here" would be easier to understand for the user.